merlin/models/tf/models/base.py

```diff
@@ -655,37 +668,35 @@ def compute_metrics(
         ----------
         prediction_outputs : PredictionOutput
             Contains properties with targets and predictions
-        training : bool
-            Flag that indicates if metrics are being computed during
-            training or evaluation
+        compiled_metrics : MetricsContainer
+            The metrics container to compute metrics on.
+            If not provided, uses self.compiled_metrics
 
         Returns
         -------
         Dict[str, tf.Tensor]
             Dict with the metrics values
         """
-
-        should_compute_metrics = self._should_compute_train_metrics_for_batch or not training
-        if should_compute_metrics:
```

Comment:
This is changing the indent of the block that runs the compute metrics. The training-time-specific logic has moved to a dedicated function `train_compute_metrics`.

And `self.compute_metrics` will re-compute metrics when it's called instead of conditionally.